frontend/routes/viewerGui/components/groups/groups.component.tsx

```diff
@@ -135,6 +135,17 @@ export class Groups extends React.PureComponent<IProps, IState> {
 	};
 
 	public groupsContainerRef = React.createRef<any>();
+	public addItemButtonRef = React.createRef<HTMLElement>();
+
+	public handleClickOutside = ({ target }) => {
+		const button = this.addItemButtonRef.current;
+		const listContainer = this.groupsContainerRef.current;
+		const containsButton = target === button || button.contains(target);
+
+		if (!listContainer.contains(target) && !containsButton) {
+			this.resetActiveGroup();
+		}
+	}
```

Comment:
I think it might be better to handle the propagation in the click handlers  instead of creating refs  unless there is another reason to use the reference to the other components.